tests/jax/systems/mappo_system_test.py

```diff
@@ -54,36 +63,36 @@ def test_except_trainer(
 
     # Optimizer.
     optimizer = optax.chain(
-        optax.clip_by_global_norm(40.0), optax.scale_by_adam(), optax.scale(-1e-4)
+        optax.clip_by_global_norm(40.0),
+        optax.adam(1e-4),
     )
 
     # Build the system
     test_full_system.build(
         environment_factory=environment_factory,
         network_factory=network_factory,
         logger_factory=logger_factory,
-        checkpoint_subpath=checkpoint_subpath,
+        experiment_path=checkpoint_subpath,
         optimizer=optimizer,
-        executor_parameter_update_period=20,
-        multi_process=False,
-        run_evaluator=True,
+        executor_parameter_update_period=1,
+        multi_process=True,
+        run_evaluator=False,
         num_executors=1,
         use_next_extras=False,
-        sample_batch_size=2,
+        sample_batch_size=5,
+        checkpoint=True,
+        nodes_on_gpu=[],
+        lp_launch_type=lp.LaunchType.TEST_MULTI_THREADING,
     )
 
-    (
-        data_server,
-        parameter_server,
-        executor,
-        evaluator,
-        trainer,
-    ) = test_full_system._builder.store.system_build
+    (trainer_node,) = test_full_system._builder.store.program._program._groups[
+        "trainer"
+    ]
+    trainer_node.disable_run()
+    test_address_builder.bind_addresses([trainer_node])
 
-    assert isinstance(executor, acme.core.Worker)
+    test_full_system.launch()
+    trainer_run = trainer_node.create_handle().dereference()
 
-    # Step the executor
-    executor.run_episode()
-
-    # Step the trainer
-    trainer.step()
+    for _ in range(2):
```

Comment:
Also, I get that we matched the way we were doing so with the TF system tests but is it not necessary to test more than the training node/ train step? Like for example, if other nodes in the system are working as expected?

Reply:
Discussed in a meeting. This smoke test check if all components work together by validating with the train step function. Other integration tests will test the other nodes